cat_merge/file_utils.py

```diff
@@ -47,6 +47,45 @@ def write_tar(tar_path: str, files: List[str], delete_files=True):
             os.remove(file)
 
 
+def read_tar_dfs(tar: tarfile.TarFile, type_name, add_provided_by: bool = True) -> List[pd.DataFrame]:
+    dataframes = []
+    for member in tar.getmembers():
+        if member.isfile() and type_name in member.name:
+            dataframes.append(read_tar_df(tar.extractfile(member), provided_by = member.name))
+    return dataframes
+
+
+def read_tar_df(file: tarfile.TarInfo, provided_by: str = None, add_provided_by: bool = True) -> pd.DataFrame:
+    df = pd.read_csv(file, sep="\t", dtype="string", lineterminator="\n", quoting=csv.QUOTE_NONE, comment='#')
+
+    if provided_by != None:
+        df["provided_by"] = provided_by
+    return df
+
+
+def read_kg(archive_path: str,
+            add_provided_by: bool = True,
+            # dangling_edges: bool = True,
+            # dangling_edges_path: str = None,
+            nodes_file_name: str = None,
+            edges_file_name: str = None):
+    if not os.path.exists(archive_path):
+        raise FileNotFoundError
+    # if dangling_edges is not None and not os.path.exists(dangling_edges_path):
+    #     raise FileNotFoundError
+
+    # iterate over files in tar, pull _nodes and _edges
+    tar = tarfile.open(archive_path, "r:*")
+    nodes_name = nodes_file_name or "_nodes"
+    edges_name = edges_file_name or "_edges"
+
+    # read into pandas and return a MergedKG instance
+    node_dfs = read_tar_dfs(tar, nodes_name, add_provided_by=add_provided_by)
+    edge_dfs = read_tar_dfs(tar, edges_name, add_provided_by=add_provided_by)
+    kg = merge_kg(node_dfs=node_dfs, edge_dfs=edge_dfs) # requires import of merge_kg form kg_utils, ok?
```

Comment:
Oh! this can actually be way easier! 

No need to re-run the merge, more like read the nodes, edges and dangling_edges into dataframes and then just instantiate a new MergedKG passing in the dataframes.

Reply:
Yes, I agree. I'll mark this as draft and roll both of the above proposed changes into a new commit.